### PR TITLE
fix: protobuf 2gb limit when checking onnx

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/4553

needed for big matrices in llms

### CODE REVIEW

**Comment by bcm-at-zama** [comment]
Don't want to contribute to upstream onnx?

File: src/concrete/ml/onnx/onnx_utils.py [hunk 1]
```diff
@@ -566,3 +570,43 @@ def remove_initializer_from_input(model: onnx.ModelProto):  # pragma: no cover
...
+                    save_as_external_data=True,
+                    all_tensors_to_one_file=True,
+                    location=external_data_path.name,
+                )
+
+                # Check the model using the file path
+                onnx.checker.check_model(str(model_path))
+        else:
+            # If it's a different error, re-raise it
+            raise
```
Comment (RomanBredehoft) [inline src/concrete/ml/onnx/onnx_utils.py:612]: you meant to write `raise e` right ?

**Review by RomanBredehoft — APPROVED** [review]
thanks !

File: src/concrete/ml/onnx/onnx_utils.py (hunk 1 above)
Comment (jfrery) [reply src/concrete/ml/onnx/onnx_utils.py:612]: this raise should raise the whole traceback. But I am not going to have coverage here...

**Comment by jfrery** [comment]
> Don't want to contribute to upstream onnx?

Not sure. I am following the work around they mention on onnx repos for such errors. I guess they would have implemented it if they really wanted to.

**Comment by bcm-at-zama** [comment]
Got it thx

**Review by RomanBredehoft — APPROVED** [review]

**Pull request merged**